server/src/eslintServer.ts

```diff
@@ -1142,5 +1145,33 @@ messageQueue.registerRequest(ExecuteCommandRequest.type, (params) => {
 	}
 });
 
+messageQueue.registerRequest(DocumentFormattingRequest.type, (params) => {
+	let textDocument: VersionedTextDocumentIdentifier = {
+		uri: params.textDocument.uri,
+		version: documents.get(params.textDocument.uri).version
+	};
+	return computeAllFixes(textDocument);
+});
+messageQueue.registerRequest(DocumentRangeFormattingRequest.type, (params) => {
+	let textDocument: VersionedTextDocumentIdentifier = {
+		uri: params.textDocument.uri,
+		version: documents.get(params.textDocument.uri).version
+	};
+	let edits = computeAllFixes(textDocument);
+	if (!edits) {
+		return [];
+	}
+	let rangedEdits = edits.filter((edit) => {
+		if (params.range.start.line < edit.range.start.line && edit.range.start.line < params.range.end.line) {

```

Comment:
Is this line correct? Consider this instead:
```suggestion
		if (params.range.start.line <= edit.range.start.line && edit.range.end.line <= params.range.end.line) {
```